### PR TITLE
feat(profiling): first draw step on CPU chart

### PR DESCRIPTION
**Pull request by JonasBa**
Currently only draws helper lines from 0-100% in 25% intervals. I'm opening this because the next few PR's will require a tiny refactor in CanvasView that I want to isolate. 

The changes here are pretty much the same boilerplate we use for other render steps, we add a model, the corresponding renderer and wire it up.

I will follow up with a stacked PR that addresses the @TODO comments and fixes configView. I disabled it while I worked on it so that the view was always zoomed out

### CODE REVIEW

File: static/app/components/profiling/flamegraph/collapsibleTimeline.tsx [hunk 1]
```diff
@@ -71,7 +71,7 @@ export function CollapsibleTimelineLoadingIndicator({size}: {size?: number}) {
 const CollapsibleTimelineContainer = styled('div')`
   position: relative;
   width: 100%;
-  height: 100%;
+  height: calc(100% - 20px);
```
Comment (JonasBa) [inline static/app/components/profiling/flamegraph/collapsibleTimeline.tsx:74]: Without this, the chart is slightly higher than the timeline as the label height is not accounted for

File: static/app/components/profiling/flamegraph/flamegraph.tsx [hunk 2]
```diff
@@ -289,6 +294,20 @@ function Flamegraph(): ReactElement {
...
+    }
+
+    return new FlamegraphChart(
+      Rect.From(flamegraph.configSpace),
+      profileGroup.measurements?.cpu_usage_0 ?? {
+        unit: 'percentage',
+        values: [],
+      }
+    );
+  }, [profileGroup.measurements?.cpu_usage_0, flamegraph.configSpace, hasCPUChart]);
```
Comment (JonasBa) [inline static/app/components/profiling/flamegraph/flamegraph.tsx:309]: Intentionally pulls just the data for 1 core as I test it.

File: static/app/utils/profiling/canvasView.tsx [hunk 3]
```diff
@@ -44,7 +44,7 @@ export class CanvasView<T extends {configSpace: Rect}> {
     this.model = model;
     this.canvas = canvas;
     this.depthOffset = options.depthOffset ?? 0;
-    this.barHeight = options.barHeight ? options.barHeight * window.devicePixelRatio : 0;
```
Comment (JonasBa) [inline static/app/utils/profiling/canvasView.tsx:47]: This was causing undefined behavior when we initialize the view as we were dividing by 0

File: static/app/components/profiling/flamegraph/flamegraphLayout.tsx [hunk 4]
```diff
@@ -264,10 +264,10 @@ const FlamegraphGrid = styled('div')<{
   width: 100%;
   grid-template-rows: ${({layout}) =>
     layout === 'table bottom'
-      ? 'auto auto auto 1fr'
+      ? 'auto auto auto auto 1fr'
```
Comment (Zylphrex) [inline static/app/components/profiling/flamegraph/flamegraphLayout.tsx:267]: Will this cause issues with platforms that don't have CPU charts?

Comment (JonasBa) [reply static/app/components/profiling/flamegraph/flamegraphLayout.tsx:267]: I tested it and seems like it just doesn't do anything if the element that takes the slot is missing. Iirc UI frames falls into the same bucket. I'll test again to make sure

File: static/app/utils/profiling/flamegraphChart.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    this.domains.y[1] = 100;
+    this.configSpace = configSpace.withHeight(this.domains.y[1] - this.domains.y[0]);
+    // this.configSpace = configSpace.withHeight(this.domains.y[1] - this.domains.y[0] + 10).withY(this.domains.y[0] - 10);
+    // new Rect(
+    //   ...this.domains.x,
+    //   this.domains.x[1] - this.domains.x[0],
+    //   this.domains.y[1] - this.domains.y[0]
+    // ) ?? Rect.Empty();
```
[the remark this answers is not in the document]
Comment (JonasBa) [reply static/app/utils/profiling/flamegraphChart.tsx]: This was removed

File: static/app/utils/profiling/flamegraphChart.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
+import {Rect} from 'sentry/utils/profiling/speedscope';
+
+import {makeFormatter} from './units/units';
+
+interface Serie {
```
Comment (Zylphrex) [inline static/app/utils/profiling/flamegraphChart.tsx:5]: The singular of series is still series.

File: static/app/utils/profiling/flamegraphChart.tsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+    for (let i = 0; i < measurement.values.length; i++) {
+      const m = measurement.values[i];
+
+      // Track and update Y max and min
+      if (m.value > this.domains.y[1]) {
+        this.domains.y[1] = m.value;
+      }
+      if (m.value < this.domains.y[0]) {
+        this.domains.y[0] = m.value;
```
Comment (Zylphrex) [inline static/app/utils/profiling/flamegraphChart.tsx:38]: We can test this but it might be better to always start the y axis for the CPU chart at 0.

**Review by Zylphrex — APPROVED** [review]

File: static/app/utils/profiling/flamegraphChart.tsx (hunk 7 above)
Comment (JonasBa) [reply static/app/utils/profiling/flamegraphChart.tsx:38]: Yeah, we'll test all of this. With the initial data I was seeing, all of it was +-1% so it was just one flat line. But yeah, we'll start with 0-100 all the time

Comment (JonasBa) [reply static/app/utils/profiling/flamegraphChart.tsx:38]: Just checked, the data is initialized to 0 so unless SDKs report < 0, this wont ever kick in

**Pull request merged**